atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableCells.java

```diff
@@ -151,12 +152,17 @@ SweepBatch getBatchForPartition(ShardAndStrategy shardStrategy, long partitionFi
     }
 
     private Multimap<Long, WriteInfo> getBatchOfWrites(SweepableCellsTable.SweepableCellsRow row,
-            PeekingIterator<Map.Entry<Cell, Value>> resultIterator) {
+            PeekingIterator<Map.Entry<Cell, Value>> resultIterator, long sweepTs) {
         Multimap<Long, WriteInfo> writesByStartTs = HashMultimap.create();
         while (resultIterator.hasNext() && writesByStartTs.size() < SweepQueueUtils.SWEEP_BATCH_SIZE) {
             Map.Entry<Cell, Value> entry = resultIterator.next();
             SweepableCellsTable.SweepableCellsColumn col = computeColumn(entry);
-            writesByStartTs.putAll(getTimestamp(row, col), getWrites(row, col, entry.getValue()));
+            long startTs = getTimestamp(row, col);
+            if (knownToBeCommittedAfterSweepTs(startTs, sweepTs)) {
+                writesByStartTs.put(startTs, getWriteInfo(startTs, entry.getValue()));
```

Comment:
You've mentioned this in the PR description - but could be worth explaining this. Something like maybe `In this case we will stop here in getTimestampsToSweepDescendingAndCleanupAborted() anyway, so there's no point getting more values`.

An uninitiated but astute reader may wonder why you're doing this if you have something like (write at 80, commit at 105) and (write at 83, commit at 96) on different cells in the same shard, with an immutable timestamp of 100.